tests/test_fft.py

```diff
@@ -47,55 +47,90 @@ def create_spmd_array(global_shape, pdims):
+def test_fft(pdims, global_shape):
+
+  yz_slab = pdims[1] == 1
 
   print("*" * 80)
-  print(f"Testing with pdims {pdims}")
+  print(f"Testing with pdims {pdims} and global shape {global_shape}")
 
-  global_shape = (29 * size, 19 * size, 17 * size
-                 )  # These sizes are prime numbers x size of the pmesh
   global_array, mesh = create_spmd_array(global_shape, pdims)
 
   # Perform distributed FFT
   with mesh:
-    karray = jaxdecomp.fft.pfft3d(global_array)
+    karray = jaxdecomp.fft.pfft3d(global_array, yz_slab=yz_slab)
     # Perform inverse FFT
-    rec_array = jaxdecomp.fft.pifft3d(karray)
+    rec_array = jaxdecomp.fft.pifft3d(karray, yz_slab=yz_slab)
 
   # Check the forward FFT
   gathered_array = multihost_utils.process_allgather(global_array, tiled=True)
   gathered_karray = multihost_utils.process_allgather(karray, tiled=True)
   gathered_rec_array = multihost_utils.process_allgather(rec_array, tiled=True)
-  jax_karray = jnp.fft.fftn(gathered_array).transpose([1, 2, 0])
-  jax_rec_array = jnp.fft.ifftn(jax_karray).transpose([2, 0, 1])
+  jax_karray = jnp.fft.fftn(gathered_array)
+
   # Check reconstructed array
   assert_allclose(
       gathered_array.real, gathered_rec_array.real, rtol=1e-7, atol=1e-7)
   assert_allclose(
       gathered_array.imag, gathered_rec_array.imag, rtol=1e-7, atol=1e-7)
-  # Check the reverse FFT
+
+  # Check the forward FFT
+  transpose_back = [2, 0, 1] if yz_slab else [1, 2, 0]
+  jax_karray_transposed = jax_karray.transpose(transpose_back)
   assert_allclose(
-      gathered_rec_array.real, jax_rec_array.real, rtol=1e-7, atol=1e-7)
+      gathered_karray.real, jax_karray_transposed.real, rtol=1e-7, atol=1e-7)
   assert_allclose(
-      gathered_rec_array.imag, jax_rec_array.imag, rtol=1e-7, atol=1e-7)
+      gathered_karray.imag, jax_karray_transposed.imag, rtol=1e-7, atol=1e-7)
 
 
-@pytest.mark.parametrize(
-    "pdims",
-    [(1, size),
-     (size, 1), pencil_1, pencil_2])  # Test with Slab and Pencil decompositions
-def test_grad(pdims):
+@pytest.mark.parametrize("global_shape",
+                         global_shapes)  # Test cubes, non-cubes and primes
+def test_yz_slab_flag(global_shape):
 
+  # When Y is distributed we have an XZ slab in input
+  # The routine is to have one 1D FFT on the contiguous X
+  # Then a single transpose from X to Y to get a YZ slab (thus the name YZ slab)
+  # In this case the routine containes only one transpose and one transpose back
+  # as apposed to the XY slab and the pencils
+  # A special flag is needed to get the right global output shape
+  # If the flag is not set, I will assert in infer_sharding_from_operands
+  # Beyond this step, it will be impossible to know if we are in a YZ slab or not
+
+  pdims = (size, 1)
   print("*" * 80)
-  print(f"Testing with pdims {pdims}")
+  print(f"Testing with pdims {pdims} and global shape {global_shape}")
+
+  global_array, mesh = create_spmd_array(global_shape, pdims)
+
+  # Perform distributed FFT
```

Comment:
I was not able to make this test do what I want
There is an assertion.
I should find a way to test with the context mesh

Reply:
Did you try putting the `try/except` inside the mesh context ?

Reply:
I have tried everything.
I will try making it work using a decorator